crates/outbound-mqtt/src/host_component.rs

```diff
@@ -0,0 +1,37 @@
+use anyhow::Context;
+use spin_app::DynamicHostComponent;
+use spin_core::HostComponent;
+
+use crate::OutboundMqtt;
+
+pub struct OutboundMqttComponent;
+
+impl HostComponent for OutboundMqttComponent {
+    type Data = OutboundMqtt;
+    fn add_to_linker<T: Send>(
+        linker: &mut spin_core::Linker<T>,
+        get: impl Fn(&mut spin_core::Data<T>) -> &mut Self::Data + Send + Sync + Copy + 'static,
+    ) -> anyhow::Result<()> {
+        spin_world::v1::mqtt::add_to_linker(linker, get)?;
```

Comment:
The v1 world is for backward compatibility.  Since old SDKs won't have MQTT anyway, you can skip it and just do v2.